lgn-messages/src/v1/mod.rs

```diff
@@ -0,0 +1,130 @@
+use std::fmt::Formatter;
+
+use mp2_v1::api::CircuitInput;
+use mp2_v1::api::PublicParameters;
+use query::MAX_NUM_COLUMNS;
+use serde::Deserialize;
+use serde::Serialize;
+
+use super::ProverType;
+use super::ToProverType;
+use crate::Proof;
+
+pub mod preprocessing;
+pub mod query;
+
+pub type ConcretePublicParameters = PublicParameters<MAX_NUM_COLUMNS>;
+pub type ConcreteCircuitInput = CircuitInput<MAX_NUM_COLUMNS>;
+
+/// Envelope for v1 messages.
+#[derive(Deserialize, Serialize)]
+pub struct Envelope {
+    /// Identifier to relate proofs with tasks.
+    pub task_id: String,
+
+    /// The task to be proved.
+    ///
+    /// Note: Deserialization of this field may fail if the mp2 changed its
+    /// representation. In that case the check for the `mp2_version` below
+    /// won't be performed. If that is an issue this struct should be converted
+    /// to an enum tagged by the mp2_version field.
+    pub task: Task,
+
+    /// The proving system version target version.
+    ///
+    /// Used to check the worker is compatible with the task.
+    pub mp2_version: String,
+}
```

Comment:
This corresponds to the [MessageEnvelope](https://github.com/Lagrange-Labs/lgn-coprocessor/blob/f48b27524eba087a8a3addaf1c6b47d7662f3996/lgn-messages/src/types/mod.rs#L45-L74) with the dead code eliminated.